model/config.go

```diff
@@ -2360,14 +2383,15 @@ func (o *Config) SetDefaults() {
 		}
 	}
 
-	o.SqlSettings.SetDefaults()
-	o.FileSettings.SetDefaults()
+	isUpdate := o.isUpdate()
```

Comment:
For safety: let's move this to the top of the function, to make it clear that we need to detect an update before writing any fields to the config.